olexandr.vakulinskiy/HW2/Task2/extended_string_class.rb

```diff
@@ -0,0 +1,16 @@
+class String
+  def valid?
+    brackets = { '(' => ')', '<' => '>', '{' => '}', '[' => ']' }
+    brackets_stack = []
+    each_char do |x|
+      brackets_stack.push(x) if brackets.key?(x)
+      if brackets.values.include?(x)
+        brackets_stack.pop if brackets[brackets_stack.last] == x
```

Comment:
Here you can do everything in the one level condition through `&&`. [Nested conditions are bad.](https://github.com/rubocop-hq/ruby-style-guide#no-nested-conditionals)
```suggestion
      brackets_stack.pop if brackets.values.include?(x) && brackets[brackets_stack.last] == x
```